test/common/http/http2/codec_impl_test.cc

```diff
@@ -965,6 +971,138 @@ TEST_P(Http2CodecImplTest, ConnectionKeepaliveJitter) {
   EXPECT_EQ(max_observed.count(), max_expected.count());
 }
 
+TEST_P(Http2CodecImplTest, DumpsStreamlessConnectionWithoutAllocatingMemory) {
+  initialize();
+  std::array<char, 1024> buffer;
+  OutputBufferStream ostream{buffer.data(), buffer.size()};
+
+  Stats::TestUtil::MemoryTest memory_test;
+  server_->dumpState(ostream, 0);
```

Comment:
nit: passing in a non-zero number of spaces here may make testing easier/more comprehensive.  suggestion: set to 2 instead of 0.

Reply:
Good idea. It now generates 2 spaces for all of these tests. Note: the `spacesForLevel` generates min(12, 2*i) spaces for i in [0, inf).